### PR TITLE
Remove words from DQA output

### PR DESCRIPTION
**Pull request by Wauplin**
Related to (unfortunately) private slack convo ([here](https://huggingface.slack.com/archives/C07SF7NFK16/p1731942920738749)). 

_originally from @Rocketknight1 :_

> Hey, for document-question-answering, the JS spec has "words", which is:
The index of each word/box pair that is in the answer
In transformers we don't output this at all. I'm not really sure how it's generated/used in the spec, since transformers already had start, end, answer, etc. 

and then 

> Update: I think the `words` output never exists because of a pipeline bug. The pipeline has two code paths in several functions, one for `VisionEncoderDecoder` models and one for everything else.
In `preprocess()`, the `VisionEncoderDecoder` path always sets [words to None](https://github.com/huggingface/transformers/blob/1c471fc3070a9179f32d147df40b760804f2ef4e/src/transformers/pipelines/document_question_answering.py#L348) which means it cannot be passed through to the output.
In `postprocess()` , the non-`VisionEncoderDecoder` path [calls](https://github.com/huggingface/transformers/blob/1c471fc3070a9179f32d147df40b760804f2ef4e/src/transformers/pipelines/document_question_answering.py#L452) `postprocess_extractive_qa`. However, this function [rewrites](https://github.com/huggingface/transformers/blob/1c471fc3070a9179f32d147df40b760804f2ef4e/src/transformers/pipelines/document_question_answering.py#L497) the `answers` dict without a `words` key
In other words, the `preprocess()` method deletes words for `VisionEncoderDecoder` models and the `postprocess()` method deletes it for everything else, so it always gets deleted! The right solution is just to remove it from the docstring and the JS spec.

---

Also to mention: DQA is only served from `transformers` in the Inference API.

### CODE REVIEW

**Comment by Wauplin** [comment]
(closing as duplicate of https://github.com/huggingface/huggingface.js/pull/1040)
